bee-network/src/service/command.rs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::PeerRelation;
+
+use libp2p::{Multiaddr, PeerId};
+use tokio::sync::mpsc;
+
+pub type CommandReceiver = mpsc::UnboundedReceiver<Command>;
+pub type CommandSender = mpsc::UnboundedSender<Command>;
+
+pub fn command_channel() -> (CommandSender, CommandReceiver) {
+    mpsc::unbounded_channel()
+}
+
+/// Describes the commands accepted by the networking layer.
+#[derive(Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum Command {
+    /// Adds a peer.
+    AddPeer {
+        /// The peer's id.
+        peer_id: PeerId,
+        /// The peer's address.
+        multiaddr: Multiaddr,
+        /// The peer's optional alias.
+        alias: Option<String>,
+        /// The relation with that peer.
+        relation: PeerRelation,
+    },
+    /// Removes a peer.
+    RemovePeer {
+        /// The peer's id.
+        peer_id: PeerId,
+    },
+    /// Connects a peer.
+    DialPeer {
```

Comment:
Feels a bit inconsistent to have `DialPeer` that connects and then `DisconnectPeer`, why not `ConnectPeer` if it actually connects it ?

Reply:
yea, I kinda agree with. From the docs there you see, that it used to be `ConnectPeer`, but I decided to follow the terminology used in `libp2p` where they talk of `dialing` instead of `connecting`. It kinda makes sense because what we actually do here is a `TryConnect` and `Dial` is a shorter, nicer term for that IMHO. However once the dialing succeeded you are in fact "connected", not just "dialed", hence breaking up that connection is correctly phrased with the term "Disconnect".